chirp/drivers/baofeng_uv17Pro.py

```diff
@@ -0,0 +1,1039 @@
+      u8 offset[6];
+      u8 unknown5[2];
+      u8 sqmode;
+      u8 unknown6[3];
+      };
+
+    #seekto 0x8000;
+    struct {
+      struct vfo_entry a;
+      struct vfo_entry b;
+    } vfo;
+
+    struct {
+      u8 squelch;
+      u8 savemode;
+      u8 vox;
+      u8 backlight;
+      u8 dualstandby;
+      u8 tot;
+      u8 beep;
+      u8 voicesw;
+      u8 voice;
+      u8 sidetone;
+      u8 scanmode;
+      u8 pttid;
+      u8 pttdly;
+      u8 chadistype;
+      u8 chbdistype;
+      u8 bcl;
+      u8 autolock;
+      u8 alarmmode;
+      u8 alarmtone;
+      u8 unknown1;
+      u8 tailclear;
+      u8 rpttailclear;
+      u8 rpttaildet;
+      u8 roger;
+      u8 unknown2;
+      u8 fmenable;
+      u8 chaworkmode:4,
+         chbworkmode:4;
+      u8 keylock;
+      u8 powerondistype;
+      u8 tone;
+      u8 unknown4[2];
+      u8 voxdlytime;
+      u8 menuquittime;
+      u8 unknown5[6];
+      u8 totalarm;
+      u8 unknown6[2];
+      u8 ctsdcsscantype;
+      ul16 vfoscanmin;
+      ul16 vfoscanmax;
+      u8 gpsw;
+      u8 gpsmode;
+      u8 unknown7[2];
+      u8 key2short;
+      u8 unknown8[2];
+      u8 rstmenu;
+      u8 unknown9;
+      u8 hangup;
+      u8 voxsw;
+      u8 gpstimezone;
+    } settings;
+
+    #seekto 0x8080;
+    struct {
+      u8 code[5];
+      u8 unknown[1];
+      u8 unused1:6,
+         aniid:2;
+      u8 dtmfon;
+      u8 dtmfoff;
+    } ani;
+
+    #seekto 0x80A0;
+    struct {
+      u8 code[5];
+      u8 name[10];
+      u8 unused:8;
+    } pttid[20];
+
+    #seekto 0x8280;
+    struct {
+      char name[16];
+    } bank_name[10];
+    """
+
+    @classmethod
+    def get_prompts(cls):
+        rp = chirp_common.RadioPrompts()
+        rp.experimental = \
+            ('This driver is a beta version.\n'
+             '\n'
+             'Please save an unedited copy of your first successful\n'
+             'download to a CHIRP Radio Images(*.img) file.'
+             )
+        rp.pre_download = _(
+            "Follow these instructions to download your info:\n"
+            "1 - Turn off your radio\n"
+            "2 - Connect your interface cable\n"
+            "3 - Turn on your radio\n"
+            "4 - Do the download of your radio data\n")
+        rp.pre_upload = _(
+            "Follow this instructions to upload your info:\n"
+            "1 - Turn off your radio\n"
+            "2 - Connect your interface cable\n"
+            "3 - Turn on your radio\n"
+            "4 - Do the upload of your radio data\n")
+        return rp
+
+    def process_mmap(self):
+        """Process the mem map into the mem object"""
+        # make lines shorter for style check.
+        self._memobj = bitwise.parse(self.MEM_FORMAT, self._mmap)
+
+    def get_settings(self):
+        """Translate the bit in the mem_struct into settings in the UI"""
+        _mem = self._memobj
+        basic = RadioSettingGroup("basic", "Basic Settings")
+        bank = RadioSettingGroup("bank", "Bank names")
```

Comment:
So I wasn't really thinking about this earlier, but putting the bank names in the settings is not really right. You should expose `has_bank` and `has_bank_names` and implement a `BankModel`. That will let you do the actual bank assignments, but it also has provisions to allow changing the name of the banks in the bank editor. You can look at some of the other radios that support bank names for examples.

Reply:
I am going to have a look at it.

Reply:
The bank names are just a list of 10 bank names. Hardly uses by the radio..... Not sure if it's worth the effort.

Reply:
Are the banks not editable? Meaning can you assign memories to banks?

Reply:
No.. you cannot assign them. The radio contains 1000 channels. The first 100 channels are in the first bank. The second 100 channels are in the second bank etc.